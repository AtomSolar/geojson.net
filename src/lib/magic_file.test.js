import parse from "./magic_file";

describe("magicFile", () => {
  test("geojson", () => {
    expect(
      parse(JSON.stringify({ type: "Point", coordinates: [0, 0] }))
    ).toEqual({
      type: "Point",
      coordinates: [0, 0]
    });
  });

  test("kml", () => {
    expect(
      parse(`<?xml version="1.0" encoding="UTF-8"?>
<kml xmlns="http://www.opengis.net/kml/2.2">
  <Placemark>
    <name>Simple placemark</name>
    <description>Attached to the ground. Intelligently places itself 
       at the height of the underlying terrain.</description>
    <Point>
      <coordinates>-122.0822035425683,37.42228990140251,0</coordinates>
    </Point>
  </Placemark>
</kml>`)
    ).toMatchSnapshot();
  });

  test("gpx", () => {
    expect(
      parse(`<?xml version="1.0"?>
<gpx version="1.1" creator="GDAL 1.11.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:ogr="http://osgeo.org/gdal" xmlns="http://www.topografix.com/GPX/1/1" xsi:schemaLocation="http://www.topografix.com/GPX/1/1 http://www.topografix.com/GPX/1/1/gpx.xsd">
<metadata><bounds minlat="10.228889000000001" minlon="-1.524722000000000" maxlat="13.520000000000000" maxlon="4.196806000000000"/></metadata>                   
<trk>
  <type>NC</type>
  <trkseg>
    <trkpt lat="10.228889" lon="2.665278">
    </trkpt>
    <trkpt lat="11.133333" lon="2.933333">
    </trkpt>
  </trkseg>
</trk>
</gpx>`)
    ).toMatchSnapshot();
  });

  // TODO: this doesn't run correctly in jest env
  test("osm", () => {
    expect(
      parse(`<osm><node id='1' lat='1.234' lon='4.321' /></osm>`)
    ).toMatchSnapshot();
  });

  test("gtfs", () => {
    expect(
      parse(`shape_id,shape_pt_lat,shape_pt_lon,shape_pt_sequence,shape_dist_traveled
A_shp,37.61956,-122.48161,1,0
A_shp,37.64430,-122.41070,2,6.8310
A_shp,37.65863,-122.30839,3,15.8765`)
    ).toMatchSnapshot();

    expect(
      parse(`stop_id,stop_name,stop_desc,stop_lat,stop_lon,zone_id,stop_url,location_type,parent_station
100014,"BEDFORD PK BL/GRAND CONCOURSE",,  40.872562, -73.888153,,,0,
100017,"PAUL AV/W 205 ST",,  40.876808, -73.889656,,,0,
100018,"PAUL AV/W MOSHOLU PY S",,  40.880341, -73.886063,,,0,
100019,"GRAND CONCOURSE/E 138 ST",,  40.813496, -73.929489,,,0,
100020,"GRAND CONCOURSE/E 144 ST",,  40.816803, -73.927956,,,0,`)
    ).toMatchSnapshot();
  });
});
