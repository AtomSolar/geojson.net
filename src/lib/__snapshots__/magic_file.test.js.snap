// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`magicFile gpx 1`] = `
Object {
  "features": Array [
    Object {
      "geometry": Object {
        "coordinates": Array [
          Array [
            2.665278,
            10.228889,
          ],
          Array [
            2.933333,
            11.133333,
          ],
        ],
        "type": "LineString",
      },
      "properties": Object {
        "type": "NC",
      },
      "type": "Feature",
    },
  ],
  "type": "FeatureCollection",
}
`;

exports[`magicFile gtfs 1`] = `
Object {
  "features": Array [
    Object {
      "geometry": Object {
        "coordinates": Array [
          Array [
            -122.48161,
            37.61956,
          ],
          Array [
            -122.4107,
            37.6443,
          ],
          Array [
            -122.30839,
            37.65863,
          ],
        ],
        "type": "LineString",
      },
      "id": "A_shp",
      "properties": Object {
        "shape_id": "A_shp",
      },
      "type": "Feature",
    },
  ],
  "type": "FeatureCollection",
}
`;

exports[`magicFile gtfs 2`] = `
Object {
  "features": Array [
    Object {
      "geometry": Object {
        "coordinates": Array [
          -73.888153,
          40.872562,
        ],
        "type": "Point",
      },
      "id": "100014",
      "properties": Object {
        "stop_id": "100014",
        "stop_name": "BEDFORD PK BL/GRAND CONCOURSE",
      },
      "type": "Feature",
    },
    Object {
      "geometry": Object {
        "coordinates": Array [
          -73.889656,
          40.876808,
        ],
        "type": "Point",
      },
      "id": "100017",
      "properties": Object {
        "stop_id": "100017",
        "stop_name": "PAUL AV/W 205 ST",
      },
      "type": "Feature",
    },
    Object {
      "geometry": Object {
        "coordinates": Array [
          -73.886063,
          40.880341,
        ],
        "type": "Point",
      },
      "id": "100018",
      "properties": Object {
        "stop_id": "100018",
        "stop_name": "PAUL AV/W MOSHOLU PY S",
      },
      "type": "Feature",
    },
    Object {
      "geometry": Object {
        "coordinates": Array [
          -73.929489,
          40.813496,
        ],
        "type": "Point",
      },
      "id": "100019",
      "properties": Object {
        "stop_id": "100019",
        "stop_name": "GRAND CONCOURSE/E 138 ST",
      },
      "type": "Feature",
    },
    Object {
      "geometry": Object {
        "coordinates": Array [
          -73.927956,
          40.816803,
        ],
        "type": "Point",
      },
      "id": "100020",
      "properties": Object {
        "stop_id": "100020",
        "stop_name": "GRAND CONCOURSE/E 144 ST",
      },
      "type": "Feature",
    },
  ],
  "type": "FeatureCollection",
}
`;

exports[`magicFile kml 1`] = `
Object {
  "features": Array [
    Object {
      "geometry": Object {
        "coordinates": Array [
          -122.0822035425683,
          37.42228990140251,
          0,
        ],
        "type": "Point",
      },
      "properties": Object {
        "description": "Attached to the ground. Intelligently places itself 
       at the height of the underlying terrain.",
        "name": "Simple placemark",
      },
      "type": "Feature",
    },
  ],
  "type": "FeatureCollection",
}
`;

exports[`magicFile osm 1`] = `undefined`;
